File: 백준/Gold/5430. AC/AC.js
'use strict';
const path = process.platform === 'linux' ? '/dev/stdin' : './input.txt';
let input = require('fs').readFileSync(path).toString().trim().split('\n');

const [T, ...testCase] = input;
const result = [];

for (let i = 0; i < T * 3; i += 3) {
  let isReverse = false;
  let isError = false;
  const p = testCase[i];
  const n = Number(testCase[i + 1]);
  let array = JSON?.parse(testCase[i + 2]);

  for (let command of p) {
    if (command === 'R') {
      isReverse = !isReverse;
    } else if (command === 'D') {
      if (array.length === 0) {
        isError = true;
        break;
      }
      isReverse ? array.pop() : array.shift();
    }
  }
  isError ? result.push('error') : result.push(JSON.stringify(isReverse ? array.reverse() : array));
}

console.log(result.join('\n'));

File: 백준/Gold/5430. AC/AC.py
import sys
from collections import deque 

input = sys.stdin.readline

T = int(input())

results = []
for _ in range(T):
  commands = input().strip()
  n = int(input())
  arr_str = input().strip()
  if arr_str == '[]':
    dq = deque()
  else:
    dq = deque(list(map(int, arr_str.strip('[]').split(','))))

  is_reverse = False
  is_error = False     
  for command in commands:
    if command == 'R':
      is_reverse = not is_reverse
    elif command == 'D':
      if len(dq) == 0:
        is_error = True
        results.append('error')
        break
      else:
        if is_reverse:
          dq.pop()
        else:
          dq.popleft()
  if is_error == False:
    if is_reverse:
      dq.reverse()
    results.append('[' + ','.join(map(str, dq)) +']')

for result in results:
  print(result)